### PR TITLE
Adding more PDF preparers

### PR DESCRIPTION
**Pull request by sree-cfa**
Includes:
- money on hand
- medical expenses

Notes:
- Sorting is for testing purposes only

### CODE REVIEW

File: src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    RESOURCES.add("Stocks");
+  }
+
+  @Override
+  public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
+    Map<String, SubmissionField> results = new HashMap<>();
+
+    var types = (List) submission.getInputData().getOrDefault("moneyOnHandTypes[]", emptyList());
+    if (!types.isEmpty()) {
+      List<String> sortedResources = RESOURCES.stream().sorted().toList();
```
Comment (bseeger) [inline src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java:44]: I wonder if you want to do this in the static section above, with the idea of doing it once and not per call to this?

Comment (bseeger) [reply src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java:44]: Or, I'm pretty sure the ArrayList will keep the order if you add them in sort order.

File: src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    EXPENSES.put("householdMedicalExpenseAmount_wildcard_Other medical expenses", "Other medical expenses");
+  }
+
+  @Override
+  public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
+    Map<String, SubmissionField> results = new HashMap<>();
+
+    var expenses = (List) submission.getInputData().getOrDefault("householdMedicalExpenses[]", emptyList());
+    if (!expenses.isEmpty()) {
+      List<String> sortedExpenses = EXPENSES.keySet().stream().sorted().toList();
```
Comment (bseeger) [inline src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java:39]: Might be worth keeping another static class variable that contains this value - so you can just do it once overall and not on every call to this.

File: src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      int i = 1;
+      for (String expense : sortedResources) {
+        var amount = submission.getInputData().get(AMOUNT_PREFIX + expense);
```
Comment (bseeger) [inline src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java:47]: This question is for both preparers -- is there any value to sorting them?  Maybe it's not necessary?  It's not clear that sorting adds any value in walking through the resource list, but I could be missing something outside of this that makes it necessary.

File: src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java (hunk 2 above)
Comment (bseeger) [reply src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java:39]: For the sorting part - I can see why you need a list of keys here, but sort still may not be necessary.

Comment (bseeger) [reply src/main/java/org/ladocuploader/app/preparers/MedicalExpensesPreparer.java:39]: This raises an interesting question for me -- what if you wanted to sort a preparer list - like the medical one. Are you finding that if you sort and then add fields in that order they show up in that order?

**Comment by bseeger** [comment]
This probably shouldn't say "you" for the name (Household resources) : Screenshot 2023-12-22 at 10 17 21 AM

File: src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java (hunk 1 above)
Comment (sree-cfa) [reply src/main/java/org/ladocuploader/app/preparers/HouseholdResourcesPreparer.java:44]: Oh yeah thats a good point - the sorting is just for testing purposes. 

A lot of these preparers with "type, amount, freq" fields are really similar and I was doing a lot of c/p

**Review by bseeger — APPROVED** [review]
LGTM!

**Pull request merged**